jib-core/src/main/java/com/google/cloud/tools/jib/configuration/credentials/Credential.java

```diff
@@ -22,18 +22,39 @@
 /** Holds credentials (username and password). */
 public class Credential {
 
+  /**
+   * Gets a {@link Credential} configured with a username and password.
+   *
+   * @param username the username
+   * @param password the password
+   * @return a new {@link Credential}
+   */
+  public static Credential basic(String username, String password) {
```

Comment:
Are we going to be adding non-basic ways of constructing credentials?

Reply:
Maybe - we could in the future maybe add token-based credentials, but I think basic is sufficient for now.

Reply:
I'm just not sure about whether a static method here is better than a simple constructor. Just seems like an extra hurdle for someone unfamiliar with the API trying to figure out how to create a `Credential`. If we were adding more ways of creating credentials, then it maybe makes sense, but as it is right now I don't think it does.

Reply:
Makes sense